src/npc.cpp

```diff
@@ -2316,25 +2316,18 @@ void npc::process_turn()
     // TODO: Make NPCs leave the player if there's a path out of map and player is sleeping/unseen/etc.
 }
 
-std::ostream &operator<< ( std::ostream &os, npc_need need )
-{
-    switch( need ) {
-        case need_none :
-            return os << "need_none";
-        case need_ammo :
-            return os << "need_ammo";
-        case need_weapon :
-            return os << "need_weapon";
-        case need_gun :
-            return os << "need_gun";
-        case need_food :
-            return os << "need_food";
-        case need_drink :
-            return os << "need_drink";
-        case num_needs :
-            return os << "num_needs";
+const std::string need_id( npc_need need )
```

Comment:
The function can return a reference (namely, `const std::string& need_id( npc_need need )`). Otherwise, the `const` qualifier is meaningless. Also, it needs to be defined in the anonymous `namespace`.

Reply:
`need_id` is also used in `debug_menu.cpp`

Reply:
Oh, I missed that. Then I'd suggest leaving the `<<` operator intact, but renaming `need_id()` to something more location-related. Their concerns are different and probably shouldn't be coupled.